kmip/services/server/session.py

```diff
@@ -51,6 +51,9 @@ def __init__(self, engine, connection, name=None):
             kwargs={}
         )
 
+        if name is None:
+            name = threading.current_thread().name
```

Comment:
Good catch! However, this fix will pull the name from the parent thread (usually 'MainThread') and therefore allows for multiple threads to have the same name if created by the same parent thread. The `threading` module will auto name threads if `name=None` is passed to the `super(...).__init__(...)` call. Let's change the logging line below to use `self.name` instead of `name`; that should resolve the `NoneType` error.